cabal-install/Distribution/Client/Types.hs

```diff
@@ -131,16 +145,28 @@ instance HasUnitId ConfiguredId where
 instance Package (ConfiguredPackage loc) where
   packageId cpkg = packageId (confPkgSource cpkg)
 
-instance PackageFixedDeps (ConfiguredPackage loc) where
-  depends cpkg = fmap (map installedUnitId) (confPkgDeps cpkg)
+instance PackageInstalled (ConfiguredPackage loc) where
+  installedDepends = CD.flatDeps . fmap (map installedUnitId) . confPkgDeps
 
 instance HasUnitId (ConfiguredPackage loc) where
   installedUnitId = confPkgId
 
+class HasConfiguredId a where
+    configuredId :: a -> ConfiguredId
```

Comment:
Although in a sense it's obvious, it's a thing that has a ConfiguredId, it would still be useful I think to document the intention/purpose/use-cases for this class.